packages/react-components/src/components/knowledge-graph/KnowledgeGraphPanel.tsx

```diff
@@ -0,0 +1,93 @@
+import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
+import { Graph } from './graph';
+import StateManager, { useKnowledgeGraphState } from './StateManager';
+import { createKnowledgeGraphQueryClient } from './KnowledgeGraphQueries';
+import { ResponseParser } from './responseParser';
+import { getElementsDefinition } from './utils';
+
+interface KnowledgeGraphInterface {
+  kgDataSource: TwinMakerKGQueryDataModule;
+}
+const MAX_NUMBER_HOPS = 10;
+
+const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({ kgDataSource }) => {
+  const { selectedGraphNodeEntityId, setQueryResult, queryResult, clearGraphResults } = useKnowledgeGraphState();
+  const [searchTerm, setSearchTerm] = useState('');
+  //const [numOfHops, setNumOfHops] = useState(MAX_NUMBER_HOPS); // number of hops can be changed in settings which will come in the coming PRs.
+  const [elements, setElements] = useState<ElementDefinition[]>([]);
+
+  const knowledgeGraphQueryClient = useMemo(() => {
+    return createKnowledgeGraphQueryClient(kgDataSource, setQueryResult);
+  }, [kgDataSource, setQueryResult]);
+
+  const onSearchClicked = useCallback(() => {
+    if (searchTerm) {
+      knowledgeGraphQueryClient.findEntitiesByName(searchTerm);
+    }
+  }, [knowledgeGraphQueryClient, searchTerm]);
+
+  const onExploreClicked = useCallback(() => {
+    if (selectedGraphNodeEntityId) {
+      knowledgeGraphQueryClient.findRelatedEntities(selectedGraphNodeEntityId, MAX_NUMBER_HOPS);
+    }
+  }, [selectedGraphNodeEntityId, knowledgeGraphQueryClient]);
+
+  const onClearClicked = useCallback(() => {
+    clearGraphResults(true);
+  }, [clearGraphResults]);
+
+  useEffect(() => {
+    if (queryResult) {
+      console.log('queryResults: ', queryResult);
+      const { nodeData, edgeData } = ResponseParser.parse(queryResult['rows'], queryResult['columnDescriptions']);
+      setElements(getElementsDefinition([...nodeData.values()], [...edgeData.values()]));
+    } else {
+      setElements([]);
+      setSearchTerm('');
+    }
+  }, [queryResult]);
+  return (
+    <Container header={<Header variant='h3'>Knowledge Graph</Header>}>
+      <SpaceBetween direction='vertical' size='s'>
+        <SpaceBetween direction='horizontal' size='s'>
+          <Input
+            type='text'
+            value={searchTerm}
+            onChange={(e) => {
+              setSearchTerm(e.detail.value);
+            }}
+          ></Input>
+          <Button onClick={onSearchClicked}>Search</Button>
+        </SpaceBetween>
+        {/* inline styling here for testing only this will be fixed in the next PR */}
+        <Graph elements={elements} style={{ width: '1000px', height: '1000px' }} />
+        <SpaceBetween direction='horizontal' size='s'>
+          {selectedGraphNodeEntityId ? (
+            <Button onClick={onExploreClicked}>Explore</Button>
+          ) : (
+            <Button disabled onClick={onExploreClicked}>
+              Explore
```

Comment:
will consider that for sure, we have very limited text we need to localize since its all graph visualization. Header, 3 buttons and probably thats it. Will dedicate a PR for this.

Reply:
There is no costly refactor its all 4 or 5 strings, I need to get the basic functionality working first. I am not going to bother with this right now.